javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1666,7 +1666,7 @@ public U getNext() {
     @Override
     default Iterator<Seq<T>> sliding(long size, long step) {
         if (size <= 0 || step <= 0) {
-            throw new IllegalArgumentException(String.format("size: %s or step: %s not positive", size, step));
+            throw new IllegalArgumentException("size: " + size + " or step: " + step + " not positive");
```

Comment:
Could you please explain what the advantage is of removing a `format` in case of an exception?

I don't think It needs to be faster and I don't think the rest of the code is `GWT` compliant (I got 3 `Java 8` compiler errors during development only, I doubt GWT can handle it...). But I might be wrong also :).